src/main/java/org/jetbrains/research/kotlincodesmelldetector/core/distance/Entity.java

```diff
@@ -0,0 +1,28 @@
+package org.jetbrains.research.kotlincodesmelldetector.core.distance;
+
+import com.intellij.psi.SmartPsiElementPointer;
+import org.jetbrains.kotlin.psi.KtClassOrObject;
+import org.jetbrains.kotlin.psi.KtElement;
+import org.jetbrains.research.kotlincodesmelldetector.utils.PsiUtils;
+
+import java.util.Objects;
+import java.util.Set;
+
+public abstract class Entity {
```

Comment:
We decided not to have additional abstractions other Kotiln PSI. Is that class necessary?

Reply:
Of course, we can work with plain KtElement's, and check the instanceof KtFunction / KtProperty when necessary. I'm just not sure if the optimization worths the code readability lost.